AssetRipper.Library/Exporters/Audio/AudioMixerExportCollection.cs

```diff
@@ -0,0 +1,306 @@
+				}
+				effect.EnableWetMix_C244 = enableWetMix;
+				effect.Bypass_C244 = effectConstant.Bypass;
+			}
+			
+			// append an Attenuation effect to a group if it has not yet got one,
+			// as Unity doesn't store Attenuation effect if it is the last.
+			foreach (IAudioMixerGroupController group in context.groups)
+			{
+				if (!groupsWithAttenuation.Contains(group))
+				{
+					IAudioMixerEffectController effect = context.virtualFile.CreateAsset<IAudioMixerEffectController>(ClassIDType.AudioMixerEffectController);
+					effect.ObjectHideFlags = HideFlags.HideInHierarchy | HideFlags.HideInInspector;
+					PPtr_AudioMixerEffectController_ effectPPtr = new();
+					effectPPtr.CopyValues(effect.SerializedFile.CreatePPtr(effect));
+					group.Effects_C243.Add(effectPPtr);
+					AddAsset(effect);
+					
+					effect.EffectID_C244.CopyValues((GUID)UnityGUID.NewGuid());
+					effect.EffectName_C244.String = "Attenuation";
+				}
+			}
+		}
+
+		private void ProcessAudioMixerSnapshots(ref AssetsProcessingContext context)
+		{
+			for (int i = 0; i < context.mixer.Snapshots_C241.Count; i++)
+			{
+				PPtr_AudioMixerSnapshot_ snapshotPPtr = context.mixer.Snapshots_C241[i];
+				IAudioMixerSnapshotController snapshot = (IAudioMixerSnapshotController)snapshotPPtr.GetAsset(context.mixer.SerializedFile);
+				AddAsset(snapshot);
+				
+				SnapshotConstant snapshotConstant = context.constants.Snapshots[i];
+				for (int j = 0; j < snapshotConstant.Values.Length; j++)
+				{
+					if (context.indexToGuid.TryGetValue((uint)j, out GUID? valueGuid))
+					{
+						snapshot.FloatValues_C245[valueGuid] = snapshotConstant.Values[j];
+					}
+					else
+					{
+						Logger.Warning(LogCategory.Export, $"Snapshot({snapshot.Name_C245}) value #{j} has no binding parameter");
+					}
+				}
+
+				for (int j = 0; j < snapshotConstant.TransitionIndices.Length; j++)
+				{
+					uint paramIndex = snapshotConstant.TransitionIndices[j];
+					int transitionType = (int)snapshotConstant.TransitionTypes[j];
+					if (context.indexToGuid.TryGetValue(paramIndex, out GUID? paramGuid))
+					{
+						snapshot.TransitionOverrides_C245[paramGuid] = transitionType;
+					}
+					else
+					{
+						Logger.Warning(LogCategory.Export, $"Snapshot({snapshot.Name_C245}) transition #{paramIndex} has no binding parameter");
+					}
+				}
+			}
+		}
+
+		private void ProcessAudioMixer(ref AssetsProcessingContext context)
+		{
+			// generate exposed parameters
+			for (int i = 0; i < context.constants.ExposedParameterIndices.Length; i++)
+			{
+				uint paramIndex = context.constants.ExposedParameterIndices[i];
+				uint paramNameCrc = context.constants.ExposedParameterNames[i];
+				if (context.indexToGuid.TryGetValue(paramIndex, out GUID? paramGuid))
+				{
+					ExposedAudioParameter exposedParam = context.mixer.ExposedParameters_C241.AddNew();
+					exposedParam.Guid.CopyValues(paramGuid);
+					exposedParam.NameString = $"{CrcUtils.ReverseDigestAscii(paramNameCrc)}";
+				}
```

Comment:
This looks like it would generate duplicates if the user exported a second time.

Reply:
fixed